train_unet.py

```diff
@@ -0,0 +1,123 @@
+import argparse
+import enum
+import json
+import os
+
+import torch
+import numpy as np
+from torch import optim
+from torch.nn import functional as F
+from torch.utils.data import Subset
+from torchvision import transforms as trfs
+from brats import transformations
+from brats.data.datasets import read_dataset_json
+from brats.losses import DiceLoss
+from brats.metrics import DiceScore
+from brats.models import UNet3D
+from brats.data import datasets
+from brats.training.runners import run_training_epoch, run_validation_epoch
+from brats.training.stop_conditions import EarlyStopping
+from brats.training.loggers import TensorboardLogger, BestModelLogger, BestStateDictLogger, ModelLogger, \
+    StateDictsLogger, log_parameters
+
+
+class Labels(enum.IntEnum):
+    EDEMA = 0
+    NON_ENHANCING = 1
+    ENHANCING = 2
+
+
+if __name__ == '__main__':
+
+    def create_parser():
+        parser = argparse.ArgumentParser(description='Train UNet 3D.')
+        parser.add_argument('--dataset_json', type=str, required=True)
+        parser.add_argument('--division_json', type=str, required=True)
+        parser.add_argument('--log_dir', type=str, required=True)
+        parser.add_argument('--device', type=str, required=True)
+        parser.add_argument('--epochs', type=int, required=True)
+        parser.add_argument('--batch_size', type=int)
+        parser.add_argument('--patience', type=int, default=10)
+        parser.add_argument('--learning_rate', type=float, default=0.001)
+        parser.add_argument('--input_size', type=int, default=240)
+        return parser
+
+
+    parser = create_parser()
+    args = parser.parse_args()
+
+    volumes_transformations = trfs.Compose([transformations.NiftiToTorchDimensionsReorderTransformation(),
+                                            trfs.Lambda(lambda x: x[3, :, :, :]),
+                                            trfs.Lambda(lambda x: np.expand_dims(x, 0)),
+                                            trfs.Lambda(lambda x: torch.from_numpy(x)),
+                                            trfs.Lambda(
+                                                lambda x: F.pad(x, [0, 0, 0, 0, 5, 0]) if x.shape[1] % 2 != 0 else x),
+                                            trfs.Lambda(lambda x: F.interpolate(x, size=args.input_size)),
+                                            transformations.StandardizeVolume(),
+                                            trfs.Lambda(lambda x: x.float()),
+                                            ])
+    masks_transformations = trfs.Compose([trfs.Lambda(lambda x: np.expand_dims(x, 3)),
```

Comment:
I found a transformation for this in `transformations.py` -> `AddChannelDimToMaskTransformation`. The question is whether we want to keep it here (see a few comments above)

Reply:
I forgot about this one, changed.